packages/firestore/test/unit/specs/bundle_spec.test.ts

```diff
@@ -0,0 +1,349 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Query } from '../../../src/core/query';
+import { doc, path, TestSnapshotVersion, version } from '../../util/helpers';
+
+import { describeSpec, specTest } from './describe_spec';
+import { client, spec } from './spec_builder';
+import { TestBundleBuilder } from '../util/bundle_data';
+import {
+  JSON_SERIALIZER,
+  TEST_DATABASE_ID
+} from '../local/persistence_test_helpers';
+import { DocumentKey } from '../../../src/model/document_key';
+import * as api from '../../../src/protos/firestore_proto_api';
+import { Value } from '../../../src/protos/firestore_proto_api';
+import { toVersion } from '../../../src/remote/serializer';
+
+interface TestBundleDocument {
+  key: DocumentKey;
+  readTime: TestSnapshotVersion;
+  createTime?: TestSnapshotVersion;
+  updateTime?: TestSnapshotVersion;
+  content?: api.ApiClientObjectMap<Value>;
```

Comment:
The existing spec tests use the "user facing types" rather than the Value proto to encode the types. This means that we would store `{ foo: 1 }` rather than `{ foo: { integerValue: 1 }}`. I think that you will find that following this pattern will greatly simplify replaying these tests on Android and iOS.

Reply:
Done.